external-crates/move/crates/move-analyzer/src/symbols.rs

```diff
@@ -500,6 +507,105 @@ fn struct_type_list_to_ide_string(types: &[(Type, bool)]) -> String {
         .join(", ")
 }
 
+/// Conversions of constant values to strings is currently best-effort which is why this function
+/// returns an Option (in the worst case we will display constant name and type but no value).
+fn const_val_to_ide_string(exp: &Exp) -> Option<String> {
+    ast_exp_to_ide_string(exp)
+}
+
+fn ast_exp_to_ide_string(exp: &Exp) -> Option<String> {
+    use UnannotatedExp_ as UE;
+    let sp!(_, e) = &exp.exp;
+    match e {
+        UE::Constant(mod_ident, name) => Some(format!("{mod_ident}::{name}")),
+        UE::Value(v) => Some(ast_value_to_ide_string(v)),
+        UE::Vector(_, _, _, exp) => ast_exp_to_ide_string(exp).map(|s| format!("[{s}]")),
+        UE::Block(seq) => {
+            let seq_items = seq
+                .iter()
+                .map(ast_seq_item_to_ide_string)
+                .collect::<Vec<_>>();
+            if seq_items.iter().any(|o| o.is_none()) {
+                // even if only one element cannot be turned into string, don't try displaying block content at all
+                return None;
+            }
+            Some(
+                seq_items
+                    .into_iter()
+                    .map(|o| o.unwrap())
+                    .collect::<Vec<_>>()
+                    .join(", "),
+            )
+        }
+        UE::ExpList(list) => {
+            let items = list
+                .iter()
+                .map(|i| match i {
+                    ExpListItem::Single(exp, _) => ast_exp_to_ide_string(exp),
+                    ExpListItem::Splat(_, exp, _) => ast_exp_to_ide_string(exp),
+                })
+                .collect::<Vec<_>>();
+            if items.iter().any(|o| o.is_none()) {
+                // even if only one element cannot be turned into string, don't try displaying expression list at all
+                return None;
+            }
+            Some(
+                items
+                    .into_iter()
+                    .map(|o| o.unwrap())
+                    .collect::<Vec<_>>()
+                    .join(", "),
+            )
+        }
+        UE::UnaryExp(op, exp) => {
+            let Some(s) = ast_exp_to_ide_string(exp) else {
+                return None;
+            };
+            Some(format!("{op}{s}"))
```

Comment:
I suppose you could `ast_exp_to_ide_string(exp).map(|s| format!("{op}{s}"))` for something shorter but I'm not convinced it's that much better. Just musing in case you prefer that.